modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/upgrade/v9_2_2/test/SchemaUpgradeProcessTest.java

```diff
@@ -0,0 +1,165 @@
+import com.liferay.object.field.builder.TextObjectFieldBuilder;
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.test.util.ObjectDefinitionTestUtil;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.db.partition.test.util.BaseDBPartitionTestCase;
+import com.liferay.portal.db.partition.util.DBPartitionUtil;
+import com.liferay.portal.kernel.instance.PortalInstancePool;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.service.CompanyLocalServiceUtil;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+import com.liferay.portal.vulcan.util.LocalizedMapUtil;
+
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Paulo Albuquerque
+ */
+@RunWith(Arquillian.class)
+public class SchemaUpgradeProcessTest extends BaseDBPartitionTestCase {
+
+	@BeforeClass
+	public static void setUpClass() throws Exception {
+		BaseDBPartitionTestCase.setUpClass();
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		Company company = CompanyLocalServiceUtil.addCompany(
+			null, _VIRTUAL_HOSTNAME, _VIRTUAL_HOSTNAME, _VIRTUAL_HOSTNAME, 0,
+			true, true, null, null, null, null, null, null);
+
+		_partitionName = DBPartitionUtil.getPartitionName(
+			company.getCompanyId());
+	}
+
+	@Test
+	public void testUpgrade() throws Exception {
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinition(
+				TestPropsValues.getCompanyId(), User.class.getSimpleName());
+
+		String dbTableName = StringBundler.concat(
+			objectDefinition.getDBTableName(), "x_",
+			PortalInstancePool.getDefaultCompanyId());
+
+		_createView(dbTableName);
+
+		objectDefinition = ObjectDefinitionTestUtil.publishObjectDefinition(
+			Collections.singletonList(
+				new TextObjectFieldBuilder(
+				).labelMap(
+					LocalizedMapUtil.getLocalizedMap(
+						RandomTestUtil.randomString())
+				).name(
+					"a" + RandomTestUtil.randomString()
+				).build()));
+
+		_createView(objectDefinition.getDBTableName());
+		_createView(objectDefinition.getExtensionDBTableName());
+
+		List<String> viewNames = new ArrayList<>();
+
+		_addViewNames(viewNames);
```

Comment:
I noticed that you always instantiate an `new ArrayList<>();` before calling the method `_addViewNames(viewNames);`.
Could you please instantiate this variable inside the method and make the method return it?
So, it would look like this:
`List<String> viewNames = _addViewNames(viewNames);`